packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -0,0 +1,519 @@
+ */
+export interface ServerlessClusterProps {
+  /**
+   * What kind of database to start
+   */
+  readonly engine: IClusterEngine;
+
+  /**
+   * Credentials for the administrative user
+   *
+   * @default - A username of 'admin' and SecretsManager-generated password
+   */
+  readonly credentials?: Credentials;
+
+  /**
+   * An optional identifier for the cluster
+   *
+   * @default - A name is automatically generated.
+   */
+  readonly clusterIdentifier?: string;
+
+  /**
+   * The number of days during which automatic DB snapshots are retained. Set
+   * to zero to disable backups.
+   *
+   * @default Duration.days(1)
+   */
+  readonly backupRetention?: Duration;
+
+  /**
+   * Name of a database which is automatically created inside the cluster
+   *
+   * @default - Database is not created in cluster.
+   */
+  readonly defaultDatabaseName?: string;
+
+  /**
+   * Indicates whether the DB cluster should have deletion protection enabled.
+   *
+   * @default - true if removalPolicy is RETAIN, false otherwise
+   */
+  readonly deletionProtection?: boolean;
+
+  /**
+   * Whether to enable the HTTP endpoint for an Aurora Serverless database cluster
+    *
+   * @default false
+   */
+  readonly enableHttpEndpoint?: boolean;
+
+  /**
+   * The VPC that this Aurora Serverless cluster has been created in.
+   */
+  readonly vpc: ec2.IVpc;
+
+  /**
+   * Where to place the instances within the VPC
+   *
+   * @default - the VPC default strategy if not specified.
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * Scaling configuration of an Aurora Serverless database cluster.
+   *
+   * @default - Serverless cluster is automatically paused after 5 minutes of being idle.
+   *   minimum capacity: 2 ACU
+   *   maximum capacity: 16 ACU
+   */
+  readonly scaling?: ServerlessScalingOptions;
+
+  /**
+   * The removal policy to apply when the cluster and its instances are removed
+   * from the stack or replaced during an update.
+   *
+   * @default - RemovalPolicy.SNAPSHOT (remove the cluster and instances, but retain a snapshot of the data)
+   */
+  readonly removalPolicy?: RemovalPolicy;
+
+  /**
+   * Security group.
+   *
+   * @default - a new security group is created.
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * The KMS key for storage encryption.
+   *
+   * @default - the default master key will be used for storage encryption
+   */
+  readonly storageEncryptionKey?: kms.IKey;
+
+  /**
+   * Additional parameters to pass to the database engine
+   *
+   * @default - no parameter group.
+   */
+  readonly parameterGroup?: IParameterGroup;
+
+  /**
+   * Existing subnet group for the cluster.
+   *
+   * @default - a new subnet group will be created.
+   */
+  readonly subnetGroup?: ISubnetGroup;
+}
+
+/**
+  * Interface representing a serverless database cluster.
+ */
+export interface IServerlessCluster extends IResource, ec2.IConnectable, secretsmanager.ISecretAttachmentTarget {
```

Comment:
This is usually above the `ServerlessClusterProps ` interface in the file (props should be right above the class that uses them).

Reply:
didn't know that, will change